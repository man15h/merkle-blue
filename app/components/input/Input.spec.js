import { mount, shallowMount } from '@vue/test-utils';
import MbInput from './Input';

describe('MbInput', () => {
  it('render correctly', () => {
    const wrapper = shallowMount(MbInput);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('is called', () => {
    const wrapper = mount(MbInput);
    expect(wrapper.name()).toBe('MbInput');
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  it('renders input element by default', () => {
    const wrapper = shallowMount(MbInput);
    expect(wrapper.contains('input')).toBeTruthy();
    expect(wrapper.classes()).toContain('flex');
  });
  it('render the placeholder and readonly attribute when passed', () => {
    const wrapper = shallowMount(MbInput, {
      attrs: { placeholder: 'Awesome!', readonly: true }
    });
    const target = wrapper.find('input');
    expect(target.element.getAttribute('placeholder')).toBe('Awesome!');
    expect(target.element.getAttribute('readonly')).toBe('readonly');
  });
});
