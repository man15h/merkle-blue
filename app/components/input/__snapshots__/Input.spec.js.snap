// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BcInput render correctly 1`] = `
<div class="flex flex-wrap justify-center">
  <div class="w-full my-auto text-left">
    <!---->
    <div class="flex flex-wrap items-stretch w-full relative"><input class="appearance-none flex-shrink flex-grow flex-auto
          leading-normal w-px flex-1 h-10 
          text-sm relative bg-grey-lighter
          text-grey-darkest border border-grey-lighter
          rounded py-2 px-3 leading-tight
          focus:outline-none focus:bg-white focus:border-grey"> </div>
  </div>
</div>
`;
